web/recv.js

```diff
@@ -0,0 +1,401 @@
+var Sink = function () {
```

Comment:
I thought about doing file reassembly on a webworker -- and we may need to go that route at some point. We need the reassembled bytes (and maybe the zstd loop?) on the main thread for saving to disk, but there are a few different ways to get there. This is the simplest.